openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -299,171 +302,181 @@ def _uidb36_to_uidb64(uidb36):
     return uidb64
 
 
-# pylint: disable=too-many-statements
-def password_reset_confirm_wrapper(request, uidb36=None, token=None):
-    """
-    A wrapper around django.contrib.auth.views.password_reset_confirm.
-    Needed because we want to set the user as active at this step.
-    We also optionally do some additional password policy checks.
-    """
-    # convert old-style base36-encoded user id to base64
-    uidb64 = _uidb36_to_uidb64(uidb36)
-    platform_name = {
-        "platform_name": configuration_helpers.get_value('platform_name', settings.PLATFORM_NAME)
-    }
-
-    # User can not get this link unless account recovery feature is enabled.
-    if 'is_account_recovery' in request.GET and not is_secondary_email_feature_enabled():
-        raise Http404
-
-    try:
-        uid_int = base36_to_int(uidb36)
-        if request.user.is_authenticated and request.user.id != uid_int:
-            raise Http404
-
-        user = User.objects.get(id=uid_int)
-    except (ValueError, User.DoesNotExist):
-        # if there's any error getting a user, just let django's
-        # password_reset_confirm function handle it.
-        return password_reset_confirm(
-            request, uidb64=uidb64, token=token, extra_context=platform_name
-        )
-
-    if UserRetirementRequest.has_user_requested_retirement(user):
-        # Refuse to reset the password of any user that has requested retirement.
-        context = {
-            'validlink': True,
-            'form': None,
-            'title': _('Password reset unsuccessful'),
-            'err_msg': _('Error in resetting your password.'),
-        }
-        context.update(platform_name)
-        return TemplateResponse(
-            request, 'registration/password_reset_confirm.html', context
-        )
-
-    if waffle().is_enabled(PREVENT_AUTH_USER_WRITES):
-        context = {
-            'validlink': False,
-            'form': None,
-            'title': _('Password reset unsuccessful'),
-            'err_msg': SYSTEM_MAINTENANCE_MSG,
+def extract_token_from_url(url):
+    token_search = re.search(r'/password_reset_confirm/.*/(.+?)/', url)
+    if token_search:
+        return token_search.group(1)
+
+
+class PasswordResetConfirmWrapper(PasswordResetConfirmView):
+    reset_url_token = 'set-password'
```

Comment:
I don't understand what this is for or why we're saving it like this.

Reply:
in `dispatch` of `PasswordResetConfrimView` a reset password form is rendered or user is redirected is determined by the reset password token retrieved from session

Reply:
Add documentation link here

Reply:
[link](https://github.com/django/django/blob/4f61810751751b8c5070ce038ea57e949650e9e3/django/contrib/auth/views.py) to Django's implementation

Reply:
ideally, it should, but in `django 1.11` this is done using a global constant and it's now (newer Django versions) moved in the class